src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -599,6 +617,8 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                 }
 
                 _packageCount += packages.Count;
+                _packageOnlyCount += packages.Select(p => p.PackageReference?.PackageIdentity.Id).Distinct(StringComparer.OrdinalIgnoreCase).Count();
```

Comment:
See comment https://github.com/NuGet/NuGet.Client/pull/4228/files#r698816435

Reply:
New property added to vs/nuget/restoreinformation event for PC projects. _packageCount actually includes csproj references too, so I added _packageOnlyCount which only account number of packages. Here _packageOnlyCount property is number of all packages in PC packages, i.e. number of packages in packages.config file. Somehow this event is not emitted when PC project is updated via install/update action on PMUI/PMC but emitted with VS restore action.

Reply:
`_packageCount` has duplicates. For example if 2 versions of `Costoso` is installed then it count them separately. Maybe I should fix `_packageCount` rather than creating new property? Please let me know.

Reply:
I decided to remove it and use _packageCount instead, technically if 2 different versions installed then they're different packages.